web_calendar_slot_duration/static/src/js/calendar_model.esm.js

```diff
@@ -0,0 +1,25 @@
+/** @odoo-module **/
+/* Copyright 2021 Tecnativa - Jairo Llopis
+
+ * License LGPL-3.0 or later (https://www.gnu.org/licenses/lgpl). */
+
+import {CalendarModel} from "@web/views/calendar/calendar_model";
+import {patch} from "@web/core/utils/patch";
+
+patch(CalendarModel.prototype, "WebCalendarSlotDurationCalendarModel", {
+    buildRawRecord(partialRecord, options = {}) {
+        if (
+            this.env.searchModel.context.calendar_slot_duration &&
+            partialRecord.isAllDay != true
+        ) {
+            var slot_duration = this.env.searchModel.context.calendar_slot_duration;
+            var slot_duration_parts = slot_duration.match(/(\d+):(\d+):(\d+)/);
+            const hours = parseInt(slot_duration_parts[1]);
+            const minutes = parseInt(slot_duration_parts[2]);
+            const seconds = parseInt(slot_duration_parts[3]);
+            const durationFloat = hours + minutes / 60 + seconds / 3600;
+            partialRecord.end = partialRecord.start.plus({hours: durationFloat});
```

Comment:
suggestion: You can destructure it for better reading (https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Destructuring_assignment):
```suggestion
            const slot_duration = this.env.searchModel.context.calendar_slot_duration;
            let [_s, hours, minutes, seconds] = slot_duration.match(/(\d+):(\d+):(\d+)/);
            let durationFloat = hours + minutes / 60 + seconds / 3600;
            partialRecord.end = partialRecord.start.plus({hours: durationFloat});
```